schism/compiler.ss

```diff
@@ -856,24 +1060,6 @@
     (if (null? expr*)
         '()
         (cons (apply-representation-expr (car expr*)) (apply-representation-expr* (cdr expr*)))))
-  (define (apply-representation-pred pred)
```

Comment:
I'm kind of glad to see the distinction between exprs and preds gone now. I've seen it used to do some clever codegen tricks, but I never really committed fully to it. I think you've accomplished the same tricks with the simplify pass.

Reply:
True.  Minor point though the distinction is still present implicitly -- e.g. the `eq?` icall produces a raw i32 value rather than a tagged value, and its continuation is always a conditional branch.  The compiler can treat the `eq?` icall just like other instructions, but it still has enough information to compile to e.g. `br_if` or something.